aws/resource_aws_dynamodb_table.go

```diff
@@ -738,6 +863,140 @@ func waitForDynamodbTableDeletion(conn *dynamodb.DynamoDB, tableName string, tim
 	return err
 }
 
+func waitForDynamoDbReplicaUpdateToBeCompleted(tableName string, region string, timeout time.Duration, conn *dynamodb.DynamoDB) error {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{
+			dynamodb.ReplicaStatusCreating,
+			dynamodb.ReplicaStatusUpdating,
+			dynamodb.ReplicaStatusDeleting,
+		},
+		Target: []string{
+			dynamodb.ReplicaStatusActive,
+		},
+		Timeout: timeout,
+		Refresh: func() (interface{}, string, error) {
+			result, err := conn.DescribeTable(&dynamodb.DescribeTableInput{
+				TableName: aws.String(tableName),
+			})
+			if err != nil {
+				return 42, "", err
+			}
+			log.Printf("[DEBUG] DynamoDB replicas: %s", result.Table.Replicas)
+
+			if len(result.Table.Replicas) == 0 {
+				return result, dynamodb.ReplicaStatusCreating, nil
+			}
+			// Find replica
+			var targetReplica *dynamodb.ReplicaDescription
+			for _, replica := range result.Table.Replicas {
+				if *replica.RegionName == region {
+					targetReplica = replica
+				}
+			}
+
+			if targetReplica == nil {
+				return nil, "", nil
+			}
```

Comment:
This logic can be consolidated and also exit early out of the loop of replicas when it matches.

```suggestion
			var targetReplica *dynamodb.ReplicaDescription

			for _, replica := range result.Table.Replicas {
				if aws.StringValue(replica.RegionName) == region {
					targetReplica = replica
					break
				}
			}

			if targetReplica == nil {
				return nil, dynamodb.ReplicaStatusCreating, nil
			}
```